pkg/informers/service/informer_service.go

```diff
@@ -145,3 +147,33 @@ func (s *ServiceImpl) GetNSTemplateTier(name string) (*toolchainv1alpha1.NSTempl
 	}
 	return tier, err
 }
+
+func (s *ServiceImpl) ListBannedUsersByEmail(email string) ([]toolchainv1alpha1.BannedUser, error) {
+	hashedEmail := hash.EncodeString(email)
+	r, err := labels.NewRequirement(toolchainv1alpha1.BannedUserEmailHashLabelKey, selection.Equals, []string{hashedEmail})
+	if err != nil {
+		return nil, err
+	}
+
+	ls := labels.NewSelector().Add(*r)
+	objs, err := s.informer.BannedUsers.ByNamespace(configuration.Namespace()).List(ls)
+	if err != nil {
+		return nil, err
+	}
+
+	bb := []toolchainv1alpha1.BannedUser{}
+	for _, obj := range objs {
+		unobj := obj.(*unstructured.Unstructured)
```

Comment:
I'm not sure this can happen. If I'm not mistaken it's stored internally as `*unstructured.Unstructured` (cf. https://github.com/kubernetes/client-go/blob/2666bd29867c96168c5e9429fd74fd9bfbedac8b/dynamic/dynamicinformer/informer.go#L138). Also I can not immagine how to write a test for covering this check.

The other functions in this file are using an unchecked casting like this too. I'd say that -if we find out it to be a problem- we can create a PR for fixing this in all the functions. WDYT?